tests/test_grid3d/test_deprecations.py

```diff
@@ -0,0 +1,64 @@
+import pytest
+from packaging import version
+
+from xtgeo import Grid, GridProperties
+from xtgeo import version as xtgeo_version
+
+
+@pytest.fixture
+def any_grid():
+    grd = Grid()
+    grd.create_box(dimension=(5, 5, 5))
+    return grd
+
+
+@pytest.fixture
+def any_gridprop(any_grid):
+    return any_grid.get_dz()
+
+
+@pytest.fixture
+def any_gridproperties(any_grid, any_gridprop):
+    gps = GridProperties(*any_grid.dimensions)
+    gps.append_props([any_gridprop])
+    return gps
+
+
+def test_grid_numpify_warns(any_grid):
+    if version.parse(xtgeo_version) < version.parse("2.7"):
+        pytest.skip()
+
+    with pytest.warns(DeprecationWarning, match="numpify_carrays"):
+        any_grid.numpify_carrays()
+
+    with pytest.warns(DeprecationWarning, match="get_cactnum"):
+        any_grid.get_cactnum()
+
+
+def test_grid_get_indices_warns(any_grid):
```

Comment:
Maybe your pattern is superior actually, but just throwing this out there as it could be an option:
```
@fail_if_not_removed(version_limit="3")
def test_default_values_deprecation():
    with pytest.warns(
        DeprecationWarning, match="as default values for RegularSurface is deprecated"
    ) as record:
        xtgeo.RegularSurface(**{"ncol": 5, "nrow": 3, "xinc": 25.0, "yinc": 25.0})
        assert len(record) == 1
```
for argument deprecation's (https://github.com/equinor/xtgeo/blob/928ae532a262d8b19582b623c38004d6a7ccb213/tests/test_surface/test_deprecation.py#L10) and:
```
@deprecation.fail_if_not_removed
@pytest.mark.usefixtures("setup_tmpdir")
def test_from_file_deprecation(default_surface):
    surface = xtgeo.RegularSurface(**default_surface)
    surface.to_file("my_file")
    surface.from_file("my_file")
```
where `deprecation` is used.

Reply:
I think it will fail anyways as the `deprecations` will give an `UnsupportedWarning` instead of a `DeprecatedWarning` once the version bound is exceeded. So the behavior of the decorator is still there.

Reply:
It would be good to not duplicate version bounds in tests. How about,
```
with pytest.warns(DeprecationWarning, match="...") as capture:
   ...
   assert len(capture) = 1
   assert capture[0].category != UnsupportedWarning
```
?

Reply:
Alternatively we could rely on matching the message. Once the bound is exceeded, the message changes from

"numpify_carrays is deprecated"
to
"... uses a function that should be removed. numpify_carrays is unsupported as of 3.0"